### PR TITLE
enable advanced_datapath provider by default

### PR DESCRIPTION
**Pull request by steved**
This is a breaking change and will require additional docs and a v3.0 release.

### CODE REVIEW

**Review by fraenkel — COMMENTED** [review]
so if I say enable advanced data path = true and enable network policy = true, the data path is enabled but the network policy is CALICO.
How is that different than PROVIDER_UNSPECIFIED?

**Comment by steved** [comment]
> so if I say enable advanced data path = true and enable network policy = true, the data path is enabled but the network policy is CALICO. How is that different than PROVIDER_UNSPECIFIED?

Shouldn't that combo result in?
```
  network_policy {
    provider = "PROVIDER_UNSPECIFIED"
    enabled  = false
  }
```

Network policy is unsupported for GKE dataplane v2, it's given by the new dataplane provider.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -50,10 +50,10 @@ repos:
         args:
           - '--args=--compact'
           - '--args=--quiet'
-          - '--args=--skip-check CKV_GCP_24,CKV_GCP_49,CKV_GCP_41,CKV_GCP_68,CKV_GCP_22,CKV_GCP_82,CKV_GCP_69,CKV_GCP_66,CKV_GCP_65,CKV_GCP_71,CKV_GCP_13,CKV_GCP_19,CKV_GCP_67,CKV_GCP_61,CKV_GCP_29,CKV_GCP_62,CKV_GCP_76,CKV_GCP_26,CKV_GCP_84,CKV_CIRCLECIPIPELINES_2,CKV_CIRCLECIPIPELINES_6,CKV2_GHA_1'
+          - '--args=--skip-check CKV_GCP_24,CKV_GCP_49,CKV_GCP_41,CKV_GCP_68,CKV_GCP_22,CKV_GCP_82,CKV_GCP_69,CKV_GCP_66,CKV_GCP_65,CKV_GCP_71,CKV_GCP_13,CKV_GCP_19,CKV_GCP_67,CKV_GCP_61,CKV_GCP_29,CKV_GCP_62,CKV_GCP_76,CKV_GCP_26,CKV_GCP_84,CKV_GCP_12:,CKV_CIRCLECIPIPELINES_2,CKV_CIRCLECIPIPELINES_6,CKV2_GHA_1'
```
Comment (miguelhar) [inline .pre-commit-config.yaml:53]: `CKV_GCP_12:,`  extra ` :`

**Review by miguelhar — APPROVED** [review]
enable_advanced_datapath = True enable_network_policy = True:
network_policy {
  provider = PROVIDER_UNSPECIFIED
  enabled = False
}

enable_advanced_datapath = True enable_network_policy = False:
network_policy {
  provider = PROVIDER_UNSPECIFIED
  enabled = False
}

enable_advanced_datapath = False enable_network_policy = True:
network_policy {
  provider = CALICO
  enabled = True
}

enable_advanced_datapath = False enable_network_policy = False:
network_policy {
  provider = CALICO
  enabled = False
}

**Comment by fraenkel** [comment]
```
enable_advanced_datapath = True enable_network_policy = True:
network_policy {
provider = PROVIDER_UNSPECIFIED
enabled = False
}
```
This combination is puzzling. I say enable network policy but we disable it. Not sure anyone will care but it almost seems like enable_net_pol should just be removed.

**Pull request closed without merging**